apps/frontend/core-dapp/src/features/ppo/stake/PPOStakingStore.ts

```diff
@@ -0,0 +1,133 @@
+import { BigNumber } from 'ethers'
+import { action, makeObservable, observable, reaction, runInAction } from 'mobx'
+import { ContractReturn, Factory } from 'prepo-stores'
+import { PPOStakingAbi, PPOStakingAbi__factory } from '../../../../generated/typechain'
+import { BalanceStructOutput } from '../../../../generated/typechain/PPOStakingAbi'
+import { Erc20Store } from '../../../stores/entities/Erc20.entity'
+import { RootStore } from '../../../stores/RootStore'
+
+type Stake = PPOStakingAbi['functions']['stake']
+type BalanceData = PPOStakingAbi['functions']['balanceData']
+
+// use this key to store balance in localStorage, while there's no SC
+const MOCK_KEY = 'MOCK_BALANCE_DATA'
+
+const TOKEN_SYMBOL = 'PPO_STAKING'
+
+// Storing data in localStorage - will allow us to test withdraw/startCooldown functionality
+function getMockedBalance(address: string): BalanceStructOutput | undefined {
+  const data = localStorage.getItem(MOCK_KEY)
+  if (!data) return undefined
+  const parsedData = JSON.parse(data) as { [key: string]: BalanceStructOutput }
+  return parsedData[address]
+}
+
+export class PPOStakingStore extends Erc20Store {
+  staking = false
+  stakingHash?: string
+  mockRawBalance?: BigNumber
+  constructor(root: RootStore) {
+    super({ root, tokenName: TOKEN_SYMBOL, factory: PPOStakingAbi__factory as unknown as Factory })
+    this.symbolOverride = TOKEN_SYMBOL
+    makeObservable(this, {
+      staking: observable,
+      stakingHash: observable,
+      stake: action.bound,
+      mockRawBalance: observable,
+      getBalanceData: observable,
+    })
+    this.subscribe()
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getBalanceData(...params: Parameters<BalanceData>): ContractReturn<BalanceData> {
+    // return this.call<BalanceData>('balanceData', params)
+    // Once SC is implemented uncomment above line, and remove lines below
+    const balance = getMockedBalance(params[0])
+    if (!balance) return undefined
+    return [balance]
+  }
+
+  get balanceData(): BalanceStructOutput | undefined {
+    const { address } = this.root.web3Store
+    if (!address) {
+      return undefined
+    }
+    const result = this.getBalanceData(address)
+    if (result === undefined) return undefined
+    return result[0]
+  }
+
+  async stake(amount: number): Promise<{ success: boolean; error?: string }> {
+    try {
+      runInAction(() => {
+        this.staking = true
+        this.stakingHash = undefined
+      })
```

Comment:
Correct me if I'm wrong but we shouldn't need to wrap any state changes with runInAction until we some await occurred since the function is already configured as `action.bound`
```suggestion
        this.staking = true
        this.stakingHash = undefined
```

Reply:
I think you are correct, but have to investigate more 
https://github.com/mobxjs/mobx/blob/mobx4and5/docs/refguide/api.md#enforceactions
there's a cool rule - WDYT